scripts/skills/perks/perk_rf_king_of_all_weapons.nut

```diff
@@ -22,106 +21,72 @@ this.perk_rf_king_of_all_weapons <- ::inherit("scripts/skills/skill", {
 		this.m.IsHidden = false;
 	}
 
+	function isEnabled()
+	{
+		if (this.m.IsForceEnabled)
+			return true;
+
+		if (this.getContainer().getActor().isDisarmed())
+			return false;
+
+		local weapon = this.getContainer().getActor().getMainhandItem();
+		return weapon != null && weapon.isWeaponType(::Const.Items.WeaponType.Spear);
+	}
+
 	function isHidden()
 	{
-		return this.m.IsSpent;
+		return this.m.IsSpent || !this.isEnabled();
 	}
 
 	function getTooltip()
 	{
-		local tooltip = this.skill.getTooltip();
-
-		tooltip.push({
+		local ret = this.skill.getTooltip();
+		ret.push({
 			id = 10,
 			type = "text",
 			icon = "ui/icons/special.png",
-			text = "The next Thrust or Prong attack costs [color=" + ::Const.UI.Color.PositiveValue + "]0[/color] Action Points, builds [color=" + ::Const.UI.Color.NegativeValue + "]0[/color] Fatigue but does [color=" + ::Const.UI.Color.NegativeValue + "]-" + this.m.DamageReductionPercentage + "%[/color] Damage"
+			text = ::Reforged.Mod.Tooltips.parseString("The next [Thrust|Skill+thrust] or [Prong|Skill+prong_skill] during your [turn|Concept.Turn] will target the body part with the lowest armor")
 		});
 
-		tooltip.push({
-			id = 10,
-			type = "text",
-			icon = "ui/icons/warning.png",
-			text = "[color=" + ::Const.UI.Color.PositiveValue + "]Will be lost upon switching your weapon![/color]"
-		});
-
-		return tooltip;
-	}
-
-	function isEnabled()
-	{
-		if (this.m.IsForceEnabled) return true;
-
-		local weapon = this.getContainer().getActor().getMainhandItem();
-		if (weapon == null || !weapon.isWeaponType(::Const.Items.WeaponType.Spear) || this.getContainer().getSkillsByFunction((@(s) this.m.Skills.find(s.getID()) != null).bindenv(this)).len() == 0)
-		{
-			return false;
-		}
-
-		return true;
+		return ret;
 	}
 
 	function onAfterUpdate( _properties )
 	{
-		if (this.m.IsSpent || !this.isEnabled() || !this.getContainer().getActor().isPlacedOnMap())
-		{
-			this.m.IsSpent = true;
+		if (!this.isEnabled())
 			return;
-		}
 
-		foreach (skill in this.getContainer().getSkillsByFunction((@(s) this.m.Skills.find(s.getID()) != null).bindenv(this)))
+		local perk = this;
+		foreach (skill in this.getContainer().getSkillsByFunction((@(s) perk.m.Skills.find(s.getID()) != null && s.m.ActionPointCost > 1)))
 		{
-			skill.m.ActionPointCost = 0;
-			skill.m.FatigueCostMult *= 0;
-		}
-	}
-
-	function onAffordablePreview( _skill, _movementTile )
-	{
-		if (_skill != null)
-		{
-			foreach (skill in this.getContainer().getSkillsByFunction((@(s) this.m.Skills.find(s.getID()) != null).bindenv(this)))
-			{
-				this.modifyPreviewField(skill, "ActionPointCost", 0, false);
-				this.modifyPreviewField(skill, "FatigueCostMult", 1, true);
-			}
+			skill.m.ActionPointCost -= 1;
```

Comment:
the `-1` should be made into a variable `this.m.ActionPointModifier` so it can be easier modified or turned off by submods or future reforged releases.

Then this line would become:
`skill.m.ActionPointCost += this.m.ActionPointModifier;`